### ISSUE TITLE
Potential memory leak in getinmemory.c example

### ISSUE THREAD

**Issue by CrazyHackGUT** [curl/curl#2991]
https://github.com/curl/curl/blob/5bd8c389a3625e5c3eed265942d4c5b1e7e5d117/docs/examples/getinmemory.c#L45-L50
`realloc` can return `null` if he can't extend the memory. This case here envisaged, but...
`realloc` doesn't clean memory. We got a memory leak if `realloc` return `null`.

If we open `realloc` documentation, we can see:
```
A null-pointer indicates that the function failed to allocate storage, and thus the block pointed by ptr was not modified.
```

**Comment by danielgustafsson** [comment]
> On 14 Sep 2018, at 10:28, Kruzya <notifications@github.com> wrote:
> 
> https://github.com/curl/curl/blob/5bd8c389a3625e5c3eed265942d4c5b1e7e5d117/docs/examples/getinmemory.c#L45-L50
> realloc can return null if he can't extend the memory. This case here envisaged, but...
> realloc doesn't clean memory. We got a memory leak if realloc return null.
> 
> If we open realloc documentation, we can see:
> 
> A null-pointer indicates that the function failed to allocate storage, and thus the block pointed by ptr was not modified.

This is not actually a memory leak as the input pointer isn’t left dangling. The pointer which wasn’t reallocated will subsequently be freed as the main function exits (line 109) unless I’m misreading this.

**Comment by danielgustafsson** [comment]
> On 14 Sep 2018, at 10:34, Daniel Gustafsson <daniel@yesql.se> wrote:
> 
>> On 14 Sep 2018, at 10:28, Kruzya <notifications@github.com> wrote:
>> 
>> https://github.com/curl/curl/blob/5bd8c389a3625e5c3eed265942d4c5b1e7e5d117/docs/examples/getinmemory.c#L45-L50
>> realloc can return null if he can't extend the memory. This case here envisaged, but...
>> realloc doesn't clean memory. We got a memory leak if realloc return null.
>> 
>> If we open realloc documentation, we can see:
>> 
>> A null-pointer indicates that the function failed to allocate storage, and thus the block pointed by ptr was not modified.
> 
> This is not actually a memory leak as the input pointer isn’t left dangling. The pointer which wasn’t reallocated will subsequently be freed as the main function exits (line 109) unless I’m misreading this.

No, I’m wrong and you’re right (as @bagder pointed out on IRC), that is indeed a leak. Do you wan’t to submit a PR with a fix for this example?

**Comment by CrazyHackGUT** [comment]
Ok, I create PR later.

**Comment by danielgustafsson** [comment]
> On 14 Sep 2018, at 23:31, Kruzya <notifications@github.com> wrote:
> 
> Ok, I create PR later.

Thanks!

**Comment by CrazyHackGUT** [comment]
I missed one time and created pull also in my fork :rofl: 
Just ignore him in this issue.

**Comment by bagder** [comment]
Fixed in 23524bf85b887a
